Adding module Lwt added to ocaml-gapi (#1)
besport/ocaml-gapi: README.md

        scopes are not returned.

An `authResponse` object have sevral value accesible with this library:

#### id_token

Accessible with `Gapi.id_token`, this "string" value represent the ID
token granted.

[...TODO add other authResponse values...]

## How to use a `googleAuth` result

Like said above, the function `Gapi.auth2_init` return a `googleAuth`,
that type can be used with other function in order to use function that
are linked to the object.

### `Gapi.then_`

This is the main use of a `googleAuth`, this function take 2 "callbacks"
in parameters: "onInit" and "onError". It returns a promise that will
calls the "onInit" function when the `GoogleAuth` object is fully
initialized or the "onError" function if an error is raised while
initializing.

Example:

```Ocaml
Gapi.then_ init
           ~onInit:(fun auth -> wakeup (Ok auth))
           ~onError:(fun msg -> wakeup (Error msg))
```

## Module `Lwt`

This module `Lwt` provide alternatives implementations of `Gapi`
functions that return a "promise" of a result, under the forme of "_
Lwt.t" objects.

[Learn more about Lwt](https://ocsigen.org/lwt/latest/manual/manual)

### `Gapi.Lwt.load`

> This version of the `load` function can't receive a "callback" argument,
> because it is used in order to "waikeup" a thread in a "wait" state.

Reviewer: ```suggestion
This version of the `load` function can't receive a *callback* argument,
because it is used in order to *wakeup* a thread in a *wait* state.
```

           ~onInit:(fun auth -> wakeup (Ok auth))
           ~onError:(fun msg -> wakeup (Error msg))
```

## Module `Lwt`

> This module `Lwt` provide alternatives implementations of `Gapi`
> functions that return a "promise" of a result, under the forme of "_
> Lwt.t" objects.

Reviewer: ```suggestion
This module `Lwt` provides alternatives implementations of `Gapi`
functions that return a *promise* of a result, under the form of `_
Lwt.t` objects.
```

[Learn more about Lwt](https://ocsigen.org/lwt/latest/manual/manual)

### `Gapi.Lwt.load`
This version of the `load` function can't receive a *callback* argument,

> because it is used in order to "wakeup" a thread in a *wait* state.

Reviewer: ```suggestion
because it is used in order to *wakeup* a thread in a *wait* state.
```

### `Gapi.Lwt.then_`

> This version of the `then_` function take only one argument of type

Reviewer: ```suggestion
This version of the `then_` function takes only one argument of type
```

> This version of the `then_` function take only 1 argument of type
> `googleAuth`, the "onInit" and "onError" function are used in order to
> provide a promise of result.

Reviewer: ```suggestion
This version of the `then_` function take only one argument of type
`googleAuth`, the `onInit` and `onError` functions are used in order to
provide a promise of result.
```

> This function return an object `(Ok auth)` if the execution of `then_`
> works or an object `(Error msg)` if the execution of `then_` failed, you
> can use theses result in your programe to differenciate cas of failure
> or success.

Reviewer: ```suggestion
This function returns an object `(Ok auth)` if the execution of `then_`
works or an object `(Error msg)` if the execution of `then_` fails, you
can use this result in your program to differentiate cases of failure
or success.
```

> This function returns an object `(Ok auth)` if the execution of `then_`
> works or an object `(Error msg)` if the execution of `then_` fails, you
> can use this result in your program to differentiate cases of failure
> or success.

Reviewer: ```suggestion
This function returns an object `Ok auth` if the execution of `then_`
works or an object `Error msg` if the execution of `then_` fails, you
can use this result in your program to differentiate cases of failure
or success.
```

Example:
```Ocaml
let init = Gapi.auth2_init (Gapi.client_config ~client_id:"id" ()) in

> let%lwt auth = Gapi.Lwt.then_

Reviewer: ```suggestion
let%lwt auth = Gapi.Lwt.then_ in
```

match auth with
| Error msg ->
  print_endline ("Failed with code error: " ^ msg)
| Ok token -> (
  [...]
)
```

> ### `Gapi.Lwt.auth2_init_then_result`

Reviewer: ```suggestion
### `Gapi.Lwt.auth2_init`
```
